cpp/src/arrow/dataset/dataset_test.cc

```diff
@@ -18,36 +18,37 @@
 #include "arrow/dataset/dataset.h"
 
 #include "arrow/dataset/test_util.h"
```

Comment:
this requires parquet libraries to be built in order to build arrow.

The error when building arrow with -DARROW_PARQUET=OFF is
```
'parquet/parquet_version.h' file not found
In file included from cpp/src/arrow/dataset/test_util.h:31:
In file included from cpp/src/parquet/arrow/writer.h:25:
```